runtime/runtime-params-estimator/src/cost.rs

```diff
@@ -82,14 +82,35 @@ pub enum Cost {
     Utf8DecodingByte,
     Utf16DecodingBase,
     Utf16DecodingByte,
+
+    // Cryptographic host functions:
+    // The runtime provides host functions for sha256, keccak256, keccak512,
+    // ripemd160 hashes. Plus, there is a ECDSA signature verification host
+    // function.
+    // All these host functions are estimated by executing a transaction with a
+    // single function call in them, that just invokes the given host function.
+    // To measure the cost of additional bytes or blocks, a function call with a
+    // large argument is measured and the total cost divided by total bytes or
+    // blocks
+    /// Estimates `sha256_base`.
```

Comment:
The comment above mentions that the total cost is divided by total bytes or blocks, but it isn't clear exactly which functions are measured by a block and which ones by a byte. Might be worthwhile to add this information to each of the types?

Reply:
The clue is in the name, `Sha256Byte` + `Keccak256Byte` + `Keccak512Byte` are measured in bytes and only `Ripemd160Block` is measured in blocks.
But I guess block is rather ambiguous in this context, at least I should probably mention that it is a compression block (as opposed to blockchain-block).

On top of that, do you think it is still worth it to add the information to each variant? I kind of would like to have a conclusive doc comment on each variant. But here it just felt a bit too repetitive to me.

Reply:
I myself try to err on the side of too much documentation than too little & I haven't yet had an opportunity to see any documentation that I thought was excessive ^^

But I definitely won't object if you decide to land the documentation here as-is.

Reply:
You are right, more documentation definitely doesn't hurt here.